src/ansys/aedt/core/hfss.py

```diff
@@ -6308,6 +6308,183 @@ def wave_port(
+            return False
+
+        if not origin:
+            origin = ["0mm", "0mm", "0mm"]
+        elif not isinstance(origin, list) or len(origin) != 3:
+            self.logger.error("Origin invalid.")
+            return False
+
+        x_origin, y_origin, z_origin = self.modeler._pos_with_arg(origin)
+
+        name = self._get_unique_source_name(name, "IncPWave")
+
+        if wave_type.lower() not in ["propagating", "evanescent", "elliptical"]:
+            self.logger.error("Wave type invalid.")
+            return False
+
+        wave_type_props = {"IsPropagating": True, "IsEvanescent": False, "IsEllipticallyPolarized": False}
+
+        if wave_type.lower() == "evanescent":
+            wave_type_props["IsPropagating"] = False
+            wave_type_props["IsEvanescent"] = True
+            if not wave_type_properties:
+                wave_type_properties = [0.0, 1.0]
+            elif not isinstance(wave_type_properties, list) or len(wave_type_properties) != 2:
+                self.logger.error("Wave type properties invalid.")
+                return False
+            wave_type_props["RealPropConst"] = wave_type_properties[0]
+            wave_type_props["ImagPropConst"] = wave_type_properties[1]
+
+        elif wave_type.lower() == "elliptical":
+            wave_type_props["IsPropagating"] = False
+            wave_type_props["IsEllipticallyPolarized"] = True
+            if not wave_type_properties:
+                wave_type_properties = ["0.0deg", 1.0]
+            elif not isinstance(wave_type_properties, list) or len(wave_type_properties) != 2:
+                self.logger.error("Wave type properties invalid.")
+                return False
+            wave_type_props["PolarizationAngle"] = wave_type_properties[0]
+            wave_type_props["PolarizationRatio"] = wave_type_properties[0]
+
+        if vector_format == "Cartesian":
+            if not polarization or polarization == "Vertical":
+                polarization = [1.0, 0.0, 0.0]
+            elif polarization == "Horizontal":
+                polarization = [0.0, 1.0, 0.0]
+            elif not isinstance(polarization, list) or len(polarization) != 3:
+                self.logger.error("Polarization invalid.")
+                return False
+
+            if not propagation_vector:
+                propagation_vector = [0.0, 0.0, 1.0]
+            elif not isinstance(propagation_vector, list) or len(propagation_vector) != 3:
+                self.logger.error("Propagation vector invalid.")
+                return False
+
+            inc_wave_args = {
+                "IsCartesian": True,
+                "EoX": polarization[0],
+                "EoY": polarization[1],
+                "EoZ": polarization[2],
+                "kX": propagation_vector[0],
+                "kY": propagation_vector[1],
+                "kZ": propagation_vector[2],
+                "OriginX": x_origin,
+                "OriginY": y_origin,
+                "OriginZ": z_origin,
```

Comment:
I would create `inc_wave_args` before the `if` statement to set `"OriginX"`, `"OriginY"` and `"OriginZ"` which are shared for cartesian format and others.